composer.json

```diff
@@ -12,6 +12,9 @@
     "prefer-stable": true,
     "require": {
         "php": ">=7.0",
+        "ext-mysqli": "*",
```

Comment:
Not sure here- do we need every single one or only PDO?

Reply:
I would say either one. Mysqli or PDO.
The script wasn't adopted to PDO becaue of the small target group.

Reply:
The old check in install.sh was not very specific. Its check returned success if at least one of the above php extensions was available. If your code do not use all 3 extensions then the install.sh check may return false-positives.

Current code base uses at least PDO, as you can see in:
https://github.com/volkszaehler/volkszaehler.org/blob/master/lib/Interpreter/DataIterator.php

php_mysqlnd is a replacement for libmysql. It is used by mysqli and PDO. It is used since PHP 5.3 by default at compile time. So we should depend on it.

The only occurrence of the string "mysqli" is in:
https://github.com/volkszaehler/volkszaehler.org/blob/master/lib/Interpreter/SQL/SQLOptimizer.php
in a switch case context and is used only if db.driver is set to "mysqli" in config. It leads to same treatment as with "pdo_mysql". So a very baseline preparation for mysqli usage.
Unit test states that "mysqli support" is not implemented jet:
https://github.com/volkszaehler/volkszaehler.org/blob/master/test/ProtocolTest.php
https://github.com/volkszaehler/volkszaehler.org/blob/master/test/RawTest.php
https://github.com/volkszaehler/volkszaehler.org/blob/master/test/SensorTest.php
this renders to "partly implemented mysqli support". Means is not ready for production. So it it not the time to raise a dependency to "php mysqli extension" until it is done. Using mysqli makes sense if you want maximum performance or very specific mysql features. Both is obviously not the case in this project. Majority of the PHP developers tend to PDO. 

In a guessing game i would bet that mysqli support willl never be implemented anymore. 
I will drop this requirement with next commit, if you agree.

Reply:
By-the-way: In practice i guess it makes no difference to depend on all 3 or only on 2 (outlined above). On a raspbian you will get all 3 extension installed when running: 

    sudo apt-get install -y php-mysql

and may be on a lot of other OS, too. 
To keep the dependencies as tight at possible can be considered a one manifestation of craftsmanship - i would tend to define 2.

Reply:
This morning I took al look over the old conversations to get it right...

> The script wasn't adopted to PDO becaue of the small target group.

The other way... The script wasn't adopted to mysqli because of the small target group

We (actually I) had a problem with a hoster distributing only mysqli with their php installation. I was working on the solution on both ends. Hosting-support and vz. Andig took al look and discovered its only a small change to add  basic (!) compability, so he changed and vz was working with mysqli.
Later i got pdo_mysql from the hoster.

> If your code do not use all 3 extensions then the install.sh check may return false-positives.

This bothered me a little too, when i realized the false positive the basic mysqli-support was already working. I wouldn't delete it complete. But i also didn't mind to check more spezific for extenions. Part of because we recommend use of Rpi with our prepared image and working php-installlation.

> To keep the dependencies as tight at possible can be considered as one manifestation of craftsmanship - i would tend to define 2.

Ok.

Reply:
I took a look at a recent travis run:
https://travis-ci.org/volkszaehler/volkszaehler.org/builds/319093512?utm_source=github_status&utm_medium=notification
and checked the "mysql" jobs. 

https://travis-ci.org/volkszaehler/volkszaehler.org/jobs/319093517
https://travis-ci.org/volkszaehler/volkszaehler.org/jobs/319093514

as you can see, there is not testing using "mysqli", it is always done with "pdo_mysql":

    sed -i "s/'pdo_mysql'/'pdo_$DB'/" etc/volkszaehler.conf.php

No quality assurance + partly implemented mysqli support = unofficial supported => drop mysqli dependency.

I pushed the change some seconds ago.
Would be nice if someone can check why https://travis-ci.org/volkszaehler/volkszaehler.org/jobs/319610222 failed.

Reply:
Wondering...
Same branch on my own fork was tested by travis with success:
https://travis-ci.org/christian-weiss/volkszaehler.org/builds/319610156